src/modules/ejector/ejector.py

```diff
@@ -285,36 +318,47 @@ def _get_latest_exit_epoch(self, blockstamp: ReferenceBlockStamp) -> tuple[Epoch
     @lru_cache(maxsize=1)
     def _get_sweep_delay_in_epochs(self, blockstamp: ReferenceBlockStamp) -> int:
         """Returns amount of epochs that will take to sweep all validators in chain."""
+        spec = self.w3.cc.get_config_spec()
         chain_config = self.get_chain_config(blockstamp)
-        total_withdrawable_validators = self._get_total_withdrawable_validators(blockstamp)
+        current_epoch = epoch_from_slot(blockstamp.slot_number, chain_config.slots_per_epoch)
+        electra_epoch = int(spec.ELECTRA_FORK_EPOCH)
+        if self.get_consensus_version(blockstamp) in (1, 2) or current_epoch < electra_epoch:
+            return self._get_sweep_delay_in_epochs_pre_pectra(blockstamp)
```

Comment:
1. consider using branching on the top level of the methods chain
2. consider using a common method for the detection of the hardfork activation